model/atmosphere/advection/tests/advection_stencil_tests/test_compute_horizontal_tracer_flux_from_linear_coefficients_alt.py

```diff
@@ -0,0 +1,74 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import gt4py.next as gtx
+import numpy as np
+import pytest
+
+import icon4py.model.common.test_utils.helpers as helpers
+from icon4py.model.atmosphere.advection.stencils.compute_horizontal_tracer_flux_from_linear_coefficients_alt import (
+    compute_horizontal_tracer_flux_from_linear_coefficients_alt,
+)
+from icon4py.model.common import dimension as dims
+
+
+class TestComputeHorizontalTracerFluxFromLinearCoefficientsAlt(helpers.StencilTest):
+    PROGRAM = compute_horizontal_tracer_flux_from_linear_coefficients_alt
+    OUTPUTS = ("p_out_e",)
+
+    @staticmethod
+    def reference(
+        grid,
+        z_lsq_coeff_1: np.array,
+        z_lsq_coeff_2: np.array,
+        z_lsq_coeff_3: np.array,
+        distv_bary_1: np.array,
+        distv_bary_2: np.array,
+        p_mass_flx_e: np.array,
+        p_vn: np.array,
+        **kwargs,
+    ):
+        e2c = grid.connectivities[dims.E2CDim]
+        z_lsq_coeff_1_e2c = z_lsq_coeff_1[e2c]
+        z_lsq_coeff_2_e2c = z_lsq_coeff_2[e2c]
+        z_lsq_coeff_3_e2c = z_lsq_coeff_3[e2c]
+
```

Comment:
since these 3 are only used one time, I would directly use `z_lsq_coeff_n[e2c]` in the computations